benchmark/test/test_framework.py.in

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+import subprocess
+import difflib
+import json
+import typing
+import re
+import pathlib
+import sys
+
+sourcepath = pathlib.Path("@CMAKE_CURRENT_SOURCE_DIR@")
+binpath = pathlib.Path("@PROJECT_BINARY_DIR@")
+generate = False
+if len(sys.argv) > 2 and sys.argv[2] == "--generate":
+    generate = True
+
+
+denumberify_paths = [
+    "time",
+    "bandwidth",
+    "flops",
+    "components",
+    "residual_norm",
+    "rhs_norm",
+    "max_relative_norm2",
+]
+empty_string_paths = ["error"]
+empty_array_paths = [
+    "recurrent_residuals",
+    "true_residuals",
+    "implicit_residuals",
+    "iteration_timestamps",
+]
+
+
+def sanitize_json_single(key, value, sanitize_all):
+    if key in empty_string_paths and isinstance(value, str):
+        return ""
+    if key in denumberify_paths and isinstance(value, float):
+        return 1.0
+    if key in denumberify_paths and isinstance(value, typing.Dict):
+        return sanitize_json(value, True)
+    if key in empty_array_paths and isinstance(value, typing.List):
+        return []
+    return sanitize_json(value, sanitize_all)
+
+
+def sanitize_json(parsed_input, sanitize_all=False):
```

Comment:
The function name is a bit misleading, since the parsed input has been already serialised to a python object.  I would probably write a function like this 

```
def recoursive_apply(parsed_input: dict, sanitizer: collections.abc.Callable) -> dict
"""Recurses into parsed_input and applys sanitizer if the values are not dictionary or list types"""
...
```

Then the sanitizer should also know sanitzie_all was set. This way you could untangle the calls from `sanitze_json` to `sanitize_json_single` and potentially back to. `sanitize_json`

Reply:
This might be really helpful in case the sanitation needs to become more advanced, but I believe with the complexity involved in such a stateful `sanitizer` object, the current implementation is still easier to maintain. I am also not sure if we will need more complex sanitation in the forseeable future, since we basically only need get rid of a handful of individual floats, arrays of floats and implementation-dependent strings.